### PR TITLE
Read bindings

### PR DESCRIPTION
**Pull request by jbgalet**
Add basic info and linear extract bindings for MLA

### CODE REVIEW

**Comment by jbgalet** [comment]
The windows tests fail because of the inconsistency between VisualStudio and the CI:
- VisualStudio runs the executable from `$(ProjectDir)` (i.e. where the `.vcxproj` is)
- The CI runs the executable from the output directory
https://github.com/ANSSI-FR/MLA/blob/ae13ac2f032a9e11eca85b280993c08e8e63cc57/.github/workflows/test.yml#L113-L115

I propose to update the CI to match VS behavior.

File: bindings/C/src/lib.rs [hunk 1]
```diff
@@ -407,3 +490,169 @@ pub extern "C" fn mla_archive_close(archive: *mut MLAArchiveHandle) -> MLAStatus
...
+}
+
+#[allow(clippy::extra_unused_lifetimes)]
+fn _mla_roarchive_walk<'a, R: Read + Seek + 'a>(
+    config: *mut MLAConfigHandle,
+    src: R,
+    file_callback: MlaFileCalback,
+    context: *mut c_void,
+) -> MLAStatus {
+    if config.is_null() || (file_callback as *mut c_void).is_null() {
```
Comment (commial) [inline bindings/C/src/lib.rs:564]: This check could be done along the one from L545

File: bindings/C/src/lib.rs [hunk 2]
```diff
@@ -407,3 +490,169 @@ pub extern "C" fn mla_archive_close(archive: *mut MLAArchiveHandle) -> MLAStatus
...
+    }
+    let mut reader = CallbackInputRead {
+        read_callback,
+        seek_callback: None,
+        context,
+    };
+    _mla_roarchive_info(&mut reader, info_out)
+}
+
+fn _mla_roarchive_info<R: Read>(src: &mut R, info_out: *mut ArchiveInfo) -> MLAStatus {
```
Comment (commial) [inline bindings/C/src/lib.rs:642]: This check could be done along the one L631

File: bindings/C/src/lib.rs [hunk 3]
```diff
@@ -407,3 +490,169 @@ pub extern "C" fn mla_archive_close(archive: *mut MLAArchiveHandle) -> MLAStatus
...
+            0 => Ok(new_pos as u64),
+            e => Err(std::io::Error::from_raw_os_error(e)),
+        }
+    }
+}
+
+/// Open an existing MLA archive using the given duration.
+/// The caller is responsible of all security checks related to callback provided paths
+#[no_mangle]
+pub extern "C" fn mla_roarchive_walk(
```
Comment (commial) [inline bindings/C/src/lib.rs:538]: I'm not a big fan of `_walk`, as it is often associated with a *browsing* of files more than an actual extraction.

We could imagine having a `_walk` API used to inspect each file, without extracting them. So, this API could be renamed with something like `_linear_extract`, `_extract`, or something like that.
What do you think?

**Comment by commial** [comment]
Hi,
Thanks for making those changes!

Apart the few asked changes, this PR seems :+1: to me

File: bindings/C/src/lib.rs (hunk 3 above)
Comment (jbgalet) [reply bindings/C/src/lib.rs]: Indeed, renamed to `mla_roarchive_extract`.

**Comment by commial** [comment]
Thanks again for the PR!
I've been way too long to merge it.

We should add, in a future PR, the API to actually list files, and read only one file -- without any extracting on disk.
That would be useful for different use cases, especially if the binding for Python is added (#139)

**Pull request merged**